### DISCUSSION TITLE
[projmgr] Do we foresee GUI based proposal on YML specification integration ?

### DISCUSSION THREAD

**Discussion by VGRSTM** [Open-CMSIS-Pack/Open-CMSIS-Pack-Spec#121] [Q&A] [answered]
Based on [YML Input Format](https://github.com/Open-CMSIS-Pack/devtools/blob/main/tools/projmgr/docs/Manual/YML-Format.md) 
 OpenCMSIS proposal is state of the art proposal supporting quite large usage from basic to complex software projects.
On one hand what's nice.
On the other I'm questioning myself how can I promote end user assistance based on some GUI proposal. Looking for example at `defines` or `add_paths` statements you can set at so many places ... plus combinatory related to `build-types` and `target-types` it sounds like a pain to end up with some user friendly proposal.

How do you foresee GUI mapping on top of  [YML Input Format](https://github.com/Open-CMSIS-Pack/devtools/blob/main/tools/projmgr/docs/Manual/YML-Format.md)  ? Is there any intention about ... or recommendation would be to rely on YML "plain text" edition ?

Comments welcome !

**Comment by ReinhardKeil** [comment, marked as answer]
We have not specified a GUI at this point in time, and perhaps should not do it as part of this Open-CMSIS project.

My view is: not every feature might be represented in a GUI, instead a GUI could show (with a flag, icon, etc.) that there are special settings for a `group`, `file`, `component`, etc. Such special settings could then be edited in the `yml` itself, or a GUI could offer a general field to enter it. 

The problem is: features like the above will be very rarely used, but when you really require it then the `yml` format could no longer be used.  There is however a fine balance to not overload the tools with features.
